slick/src/main/scala/slick/jdbc/SQLiteProfile.scala

```diff
@@ -97,9 +97,10 @@ trait SQLiteProfile extends JdbcProfile {
   class ModelBuilder(mTables: Seq[MTable], ignoreInvalidDefaults: Boolean)(implicit ec: ExecutionContext) extends JdbcModelBuilder(mTables, ignoreInvalidDefaults) {
     override def createColumnBuilder(tableBuilder: TableBuilder, meta: MColumn): ColumnBuilder = new ColumnBuilder(tableBuilder, meta) {
       /** Regex matcher to extract name and length out of a db type name with length ascription */
-      final val TypePattern = "^([A-Z]+)(\\(([0-9]+)\\))?$".r
+      final val TypePattern = """^([A-Z]+)(|\(([0-9]+)(|,([0-9]+))\))?$""".r
       private val (_dbType,_size) = meta.typeName match {
-        case TypePattern(d,_,s) => (d, Option(s).map(_.toInt))
+        case TypePattern(d,_,s,_,null) => (d, Option(s).map(_.toInt))  
+        case TypePattern(d,_,_,_,_)    => (d , None) //Ignore 2nd ascription since its ignored by sqlite type affinity to REAL or NUMERIC 
```

Comment:
I guess what he means is that the `16` and `8` in `DECIMAL(16,8)` are simply ignored in SQLite (it will result in `NUMERIC`). See: http://www.sqlite.org/datatype3.html#section_3_2